platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -181,6 +187,16 @@ void OfflineDatabase::migrateToVersion6() {
     transaction.commit();
 }
 
+void OfflineDatabase::migrateToVersion7() {
+    assert(db);
+    if (getPragma<int64_t>("PRAGMA auto_vacuum") != 2) {
+        // auto_vacuum might have been already set to INCREMENTAL at migrateToVersion3()
+        db->exec("PRAGMA auto_vacuum = INCREMENTAL");
+        db->exec("VACUUM");
```

Comment:
If the user has a 1GB database on verion 6, will this `VACUUM` result in a copy of 1GB data? Could this lead to upgrade taking a long time?

Reply:
This is a fair point, thanks! On the other hand assigning the `auto_vacuum` pragma to a new value is not a breaking change. So, after brainstorming with @tmpsantos and @alexshalamov, we think that we can just lazily reassign the `auto_vacuum` pragma for the existing databases even without version bumping (newly created databases will have `auto_vacuum = INCREMENTAL` from the start).

Reply:
@pozdnyakov Thanks! Could we skip `VACUUM` here or is it needed?

Reply:
it's required, so that `auto_vacuum = INCREMENTAL` gets to an action.